axon/src/proof.rs

```diff
@@ -1,227 +1,94 @@
+        Ok(ObjectProof {
+            receipt,
+            receipt_proof,
+            block,
+            state_root,
+            axon_proof,
+        })
+    }
 }
 
-impl Transaction {
-    pub fn rlp(&self) -> Bytes {
-        let mut rlp = RlpStream::new();
-        rlp.begin_unbounded_list();
-
-        match self.transaction_type {
-            // EIP-2930 (0x01)
-            Some(x) if x == U64::from(1) => {
-                rlp_opt(&mut rlp, &self.chain_id);
-                rlp.append(&self.nonce);
-                rlp_opt(&mut rlp, &self.gas_price);
-                rlp.append(&self.gas);
-
-                #[cfg(feature = "celo")]
-                self.inject_celo_metadata(&mut rlp);
-
-                rlp_opt(&mut rlp, &self.to);
-                rlp.append(&self.value);
-                rlp.append(&self.input.as_ref());
-                rlp_opt_list(&mut rlp, &self.access_list);
-                if let Some(chain_id) = self.chain_id {
-                    rlp.append(&normalize_v(self.v.as_u64(), U64::from(chain_id.as_u64())));
-                }
-            }
-            // EIP-1559 (0x02)
-            Some(x) if x == U64::from(2) => {
-                rlp_opt(&mut rlp, &self.chain_id);
-                rlp.append(&self.nonce);
-                rlp_opt(&mut rlp, &self.max_priority_fee_per_gas);
-                rlp_opt(&mut rlp, &self.max_fee_per_gas);
-                rlp.append(&self.gas);
-                rlp_opt(&mut rlp, &self.to);
-                rlp.append(&self.value);
-                rlp.append(&self.input.as_ref());
-                rlp_opt_list(&mut rlp, &self.access_list);
-                if let Some(chain_id) = self.chain_id {
-                    rlp.append(&normalize_v(self.v.as_u64(), U64::from(chain_id.as_u64())));
-                }
-            }
-            // Legacy (0x00)
-            _ => {
-                rlp.append(&self.nonce);
-                rlp_opt(&mut rlp, &self.gas_price);
-                rlp.append(&self.gas);
-
-                rlp_opt(&mut rlp, &self.to);
-                rlp.append(&self.value);
-                rlp.append(&self.input.as_ref());
-                rlp.append(&self.v);
-            }
-        }
-
-        rlp.append(&self.r);
-        rlp.append(&self.s);
-
-        rlp.finalize_unbounded_list();
-
-        let rlp_bytes: Bytes = rlp.out().freeze().into();
-        let mut encoded = Vec::new();
-        match self.transaction_type {
-            Some(x) if x == U64::from(1) => {
-                encoded.extend_from_slice(&[0x1]);
-                encoded.extend_from_slice(rlp_bytes.as_ref());
-                encoded.into()
-            }
-            Some(x) if x == U64::from(2) => {
-                encoded.extend_from_slice(&[0x2]);
-                encoded.extend_from_slice(rlp_bytes.as_ref());
-                encoded.into()
-            }
-            _ => rlp_bytes,
+impl Default for ObjectProof {
+    fn default() -> Self {
```

Comment:
you can just only initial some fields without default trait, and others with default trait can use `...Default::default()` to be automaticlly filled at one line, so there's no need to fill them with `Default::default()` one by one

Reply:
`..Default::default()` requires struct implementing `Default` trait. I had to write this because of AxonHeader and AxonBlock does not implement it.

Reply:
that's a sad story, I got it